docs/source/transaction_family_specifications/identity_transaction_family.rst

```diff
@@ -0,0 +1,217 @@
+*****************************************
+Identity Transaction Family Specification
+*****************************************
+
+Overview
+=========
+On-chain permissioning is desired for transactor key and validator key
+permissioning. This feature requires that authorized participants for each
+role be stored. Currently, the only form of identification on the network is
+the entity's public key. Since lists of public keys are difficult to manage,
+an identity namespace will be used to make managing identity easier.
+
+The identity system described here is an extensible, role-based, and
+policy-based system for defining permissions in a way which can be utilized by
+other pieces of the architecture. This includes the transactor key and
+validator key permissioning components, but in the future may also be used by
+transaction family implementations.
+
+The identity namespace:
+
+- Encompasses ways to identify participants based on public keys
+- Stores a set of permit and deny rules called "policies"
+- Stores the roles that those policies apply to
+
+Policies describe a set of identities that have permission to perform an
+action, without specifying the action. Roles describe a set of actions that can
+be performed.
+
+.. note::
+  This transaction family will serve as a reference specification
+  and implementation and may also be used in production. Different
+  implementations can be used so long as they adhere to the same addressing
+  scheme and content format for policies and roles.
+
+
+State
+=====
+
+Policies
+--------
+A policy will have a name and a list of entries. Each policy entry will have a
+type and a key list. The type will be either PERMIT_KEY or DENY_KEY and the key
+list will be a list of public keys.
+
+.. code-block:: protobuf
+
+  // Policy will be stored in a PolicyList to account for hash collisions
+  message PolicyList {
+    repeated Policy policies = 1;
+  }
+
+  message Policy {
+    enum Type {
+      PERMIT_KEY = 0;
+      DENY_KEY = 1;
+    }
+
+    message Entry {
+      // Whether this is a PERMIT_KEY or DENY_KEY entry
+      Type type = 1;
+      // This should be a public key or * to refer to all participants.
+      string key = 2;
+
+    }
+
+    // name of the policy, this should be unique.
+    string name = 1;
+
+    // The entries will be processed in order from first to last.
+    repeated Entry entries = 2;
+  }
+
+Roles
+-----
+A role will be made up of a role name and the name of the policy to be enforced
+for that role. The data will be stored in state at the address described above
+using the following protobuf messages:
+
+.. code-block:: protobuf
+
+  // Roles will be stored in a RoleList to handle state collisions. The Roles
+  // will be sorted by name.
+  message RoleList {
+    repeated Role roles = 1;
+  }
+
+  message Role{
+    // Role name
+    string name = 1;
+
+    // Name of corresponding policy
+    string policy_name = 2;
```

Comment:
According to the documentation above roles specific "a set of actions", but this implementation suggests each role is just a name given to a reference to a single policy. Is there a property missing here? Should this field be repeated? Is the initial paragraph inaccurate?

Reply:
For what it's worth, I also don't understand how a Role is useful if it is basically just an alias for a Policy? I assume there is a good reason, but perhaps it should be explained in an earlier paragraph.

Reply:
Description updated above.